touchforms/backend/touchcare.py

```diff
@@ -31,264 +34,53 @@
-            ledger.setEntry(section_id, product_id, int(amount))
-    return ledger
-
-
-class StaticIterator(IStorageIterator):
-    def __init__(self, ids):
-        self.ids = ids
-        self.i = 0
-
-    def hasMore(self):
-        return self.i < len(self.ids)
-
-    def nextID(self):
-        id = self.ids[self.i]
-        self.i += 1
-        return id
-
-
-class TouchformsStorageUtility(IStorageUtilityIndexed):
-    """
-    The TouchformsStorageUtility provides an interface for working with the case database. The mobile phone
-    uses this to populate and reference cases in the SQLite database on the Android phone. Touchforms uses HQ
-    as its "mobile database" so when populating the case universe, it calls HQ to get the case universe for
-    that particular user.
-
-    See:
-    https://github.com/dimagi/javarosa/blob/master/core/src/org/javarosa/core/services/storage/IStorageUtilityIndexed.java
-    for more information on the interface.
-    """
-
-    def __init__(self, host, domain, auth, additional_filters=None, preload=False, form_context=None):
-        self.cached_lookups = {}
-        self.form_context = form_context or {}
-
-        if self.form_context.get('case_model', None):
-            case_model = self.form_context['case_model']
-            self.cached_lookups[('case-id', case_model['case_id'])] = [case_from_json(case_model)]
-
-        self._objects = {}
-        self.ids = {}
-        self.fully_loaded = False  # when we've loaded every possible object
-        self.query_func = query_factory(host, domain, auth)
-        self.additional_filters = additional_filters or {}
-        if preload:
-            self.load_all_objects()
-        else:
-            self.load_object_ids()
+class CCInstances(InstanceInitializationFactory):
 
-    def get_object_id(self, object):
-        raise NotImplementedError("subclasses must handle this")
+    def __init__(self, sessionvars, auth, restore=None, force_sync=False):
+        self.vars = sessionvars
+        self.username = sessionvars['username']
+        self.auth = auth
+        self.sandbox = UserDataUtils.getStaticStorage(self.username)
+        self.host = sessionvars['host']
+        self.domain = sessionvars['domain']
+        self.query_func = query_factory(self.host, self.domain, self.auth, 'raw')
+        self.query_url = get_restore_url({'as': self.username + '@' + self.domain, 'version': '2.0'})
 
-    def fetch_object(self, object_id):
-        raise NotImplementedError("subclasses must handle this")
+        if force_sync or self.needs_sync():
+            self.perform_ota_restore(restore)
 
-    def load_all_objects(self):
-        raise NotImplementedError("subclasses must handle this")
+    def perform_ota_restore(self, restore=None):
 
-    def load_object_ids(self):
-        raise NotImplementedError("subclasses must handle this")
+        self.sandbox = UserDataUtils.getClearedStaticStorage(self.username)
```

Comment:
Seems odd to me to make a new instance rather than clearing the current instance. Just looked at the java file though and it seems like you only have two class methods to make use of